src/openrct2-ui/input/MouseInput.cpp

```diff
@@ -586,8 +586,14 @@ static void InputViewportDragContinue()
             // As the user moved the mouse, don't interpret it as right click in any case.
             _ticksSinceDragStart = std::nullopt;
 
-            differentialCoords.x = (viewport->zoom + 1).ApplyTo(differentialCoords.x);
-            differentialCoords.y = (viewport->zoom + 1).ApplyTo(differentialCoords.y);
+            // applying the zoom only with negative values avoids a "deadzone" effect where small positive value round to zero.
+            const bool posX = differentialCoords.x > 0;
+            const bool posY = differentialCoords.y > 0;
+            differentialCoords.x = (viewport->zoom + 1).ApplyTo(-std::abs(differentialCoords.x));
+            differentialCoords.y = (viewport->zoom + 1).ApplyTo(-std::abs(differentialCoords.y));
+            differentialCoords.x = posX ? -differentialCoords.x : differentialCoords.x;
+            differentialCoords.y = posY ? -differentialCoords.y : differentialCoords.y;
```

Comment:
Could you explain this? I don’t quite understand why we need to invert the value in all cases.

Reply:
The values are guaranteed to be negative at this point. These two lines only invert them (restore them to positive) if they were positive previously.

Reply:
Oh wait, there was a little `-` hiding just before the `std::abs`.